services/QuillLMS/app/models/redis_feed.rb

```diff
@@ -0,0 +1,65 @@
+# Intended to be subclassed.
+# RedisFeed allows an limited array of ids to be stored in redis
+# And those ids to be retrieved and converted to objects via a hydrate method
+# Usage:
+# Implemement subclass: SomeActivityFeed < RedisFeed
+# Add item: SomeActivityFeed.add(teacher_id, some_activity_identifier)
+# Retrieve objects: SomeActivityFeed.get(teacher_id) -> returns array of hash objects
+# Subclasses need to implement:
+# Required: key, limit, hydrate(ids:)
+# Optional: callback_on_add(id)
+
+class RedisFeed
+  attr_reader :redis_key, :key_id
+
+  class NotImplementedError < StandardError; end;
+
+  def initialize(key_id)
+    @key_id = key_id
+    @redis_key = "#{key}#{key_id}"
+  end
+
+  def self.add(key_id, id)
+    new(key_id).add(id)
+  end
+
+  def self.get(key_id)
+    feed = new(key_id)
+    feed.hydrate(ids: feed.ids)
+  end
+
+  # add identifiers to a redis array, limit to a certain size
+  def add(id)
+    $redis.lpush(redis_key, id)
+    $redis.ltrim(redis_key, 0, limit - 1)
+    callback_on_add(id)
+  end
+
+  # returns an array of strings
+  def ids
+    $redis.lrange(redis_key, 0, limit - 1)
+  end
+
+  # Methods defined by subclass
```

Comment:
Note, there's not a good way to define a protocol in Ruby (that I know of, open to suggestions), so I'm raising if the subclass doesn't implement required methods.